src/storage-controller/src/lib.rs

```diff
@@ -3425,15 +3461,13 @@ where
         latest_row_per_id
             .into_iter()
             .filter_map(|(key, (_, row_vec))| {
```

Comment:
This was preexisting, but why is it safe to just silently drop old rows whose keys don't extract correctly? Seems very suspicious...

Reply:
This code was added in https://github.com/MaterializeInc/materialize/pull/24531, but I don't see any discussion about it there.

Previously we only invoked this method on collections keyed by a `GlobalId`, so `Err` case here could never happen unless a collection was corrupted. Maybe this was added in anticipation of new status collections not keyed by a `GlobalId`? But then it still doesn't make sense that we `unwrap_str` here, and panic when the key is not a string.

Reply:
This particular code went away now with the `StatusHistoryDesc` refactor.